### PR TITLE
Backport cilium network policies to v1.7.x

### PR DESCRIPTION
**Pull request by kopiczko**
Towards https://github.com/giantswarm/giantswarm/issues/23014
Towards https://github.com/giantswarm/roadmap/issues/1949

It contains changes from:

- https://github.com/giantswarm/cluster-api-app/pull/132
- https://github.com/giantswarm/cluster-api-app/pull/137
- https://github.com/giantswarm/cluster-api-app/pull/140
- https://github.com/giantswarm/cluster-api-app/pull/141

### CODE REVIEW

File: helm/cluster-api/templates/crd-install/crd-ciliumnetworkpolicy.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+spec:
+  endpointSelector:
+    matchLabels:
+      app.kubernetes.io/component: {{ include "cluster-api.crdInstall" . | quote }}
+      {{- include "cluster-api.selectorLabels" . | nindent 6 }}
+  egress:
+    - toEntities:
+        - kube-apiserver
+{{- end }}
+{{- end }}
```
Comment (fiunchinho) [inline helm/cluster-api/templates/crd-install/crd-ciliumnetworkpolicy.yaml:25]: Is this file also generated? I was expecting to see it being added to the `config` folder too

Comment (kopiczko) [reply helm/cluster-api/templates/crd-install/crd-ciliumnetworkpolicy.yaml:25]: I don't think this directory is generated. It's a mess.

**Review by fiunchinho — APPROVED** [review]

**Comment by kopiczko** [comment]
Converting to draft. I'll need to cherry pick couple of commits to the master branch and then merge them back here.

**Comment by kopiczko** [comment]
Tested and works.

**Pull request merged**